File: src/main/java/ordilov/lissn/security/userinfo/OAuth2UserInfoFactory.java
package ordilov.lissn.security.userinfo;

import ordilov.lissn.member.domain.AuthProvider;
import ordilov.lissn.security.exception.OAuth2AuthenticationProcessingException;
import org.springframework.security.oauth2.core.user.OAuth2User;

public class OAuth2UserInfoFactory {
  public static OAuth2UserInfo getOAuth2UserInfo(AuthProvider provider, OAuth2User oAuth2User) {
    if (provider == AuthProvider.google) {
      return new GoogleOAuth2UserInfo(oAuth2User);
    } else {
      throw new OAuth2AuthenticationProcessingException("잘못된 로그인입니다.");
    }
  }
}

File: src/main/java/ordilov/lissn/security/userinfo/OAuth2UserInfoFactory.kt
package ordilov.lissn.security.userinfo;

import ordilov.lissn.member.domain.AuthProvider;
import ordilov.lissn.security.exception.OAuth2AuthenticationProcessingException;
import org.springframework.security.oauth2.core.user.OAuth2User;

class OAuth2UserInfoFactory {
    companion object {
        fun getOAuth2UserInfo(provider: AuthProvider, oAuth2User: OAuth2User): OAuth2UserInfo {
            if (provider == AuthProvider.google) {
                return GoogleOAuth2UserInfo(oAuth2User);
            } else {
                throw OAuth2AuthenticationProcessingException("잘못된 로그인입니다.");
            }
        }
    }
}